k2/csrc/fsa_equivalent.cc

```diff
@@ -14,9 +14,54 @@
 
 #include "k2/csrc/fsa.h"
 #include "k2/csrc/properties.h"
+#include "k2/csrc/fsa_algo.h"
 
 namespace k2 {
 
+bool IsRandEquivalent(const Fsa &a, const Fsa &b, std::size_t npath /*=100*/) {
+  // We will do `intersect` later which requires either `a` or `b` is
+  // epsilon-free, considering they should hold same set of arc labels, so both
+  // of them should be epsilon-free.
+  // TODO(haowen): call `RmEpsilon` here instead of checking.
+  if (!IsEpsilonFree(a) || !IsEpsilonFree(b)) return false;
+
+  Fsa connected_a, connected_b, valid_a, valid_b;
+  Connect(a, &connected_a);
```

Comment:
You need to check the return status of `Connect`.

Reply:
I dont's care the return value (is_acyclic) here. Even if it failed, leave `connected_a` with empty is fine here